api/src/main/java/io/smallrye/stork/Metadata.java

```diff
@@ -0,0 +1,41 @@
+package io.smallrye.stork;
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class Metadata<T extends Enum<T>> {
+
+    private EnumMap<T, Object> metatada;
+    Class<T> enumType;
+
+    public Metadata(Class<T> key) {
+        this.metatada = new EnumMap<T, Object>(key);
+        this.enumType = key;
+    }
+
+    public Metadata() {
+
```

Comment:
Why do we have this constructor? Is it to create empty metadata? In this case, I would use a specific static method and hide this constructor. 

The problem is that accesses to the metadata will throw NPE.

Reply:
Yes, I needed something for using when I don't know the actual type, a kind of default....ok, I will do it with a static method